lesson-demo/src/main/java/com/bobocode/pool/PooledDataSource.java

```diff
@@ -0,0 +1,33 @@
+package com.bobocode.pool;
+
+import lombok.SneakyThrows;
+import org.postgresql.ds.PGSimpleDataSource;
+
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingQueue;
+
+public class PooledDataSource extends PGSimpleDataSource {
+    // todo: 1. store a queue of connections (this is a pool)
+    // todo: 2. initialize a datasource with 10 physical connection
+    // todo: 3. override method getConnection so it uses a pool
+    private BlockingQueue<Connection> pool;
+
+    @SneakyThrows
+    @Override
+    public Connection getConnection() {
+        if (pool == null) {
+            initializePool();
+        }
+        return pool.take();
+    }
+
+    private void initializePool() throws SQLException, InterruptedException {
```

Comment:
And the second thing is that your initialisation logic, as well as underlying Queue are **not thread safe**. What will happen if two threads will call `getConnection` at the same time, when `pool == null`?